ami/main.tf

```diff
@@ -76,19 +79,61 @@ variable "virtualization_type" {
   }
 }
 
-data "external" "ami" {
-  program = [
-    "${path.module}/ami.sh",
-    path.module,
-    var.release,
-    var.region,
-    var.system,
-    var.virtualization_type,
-  ]
+locals {
+  root_device_type    = endswith(var.virtualization_type, "ebs") ? "ebs" : "instance-store"
+  virtualization_type = startswith(var.virtualization_type, "hvm") ? "hvm" : "paravirtual"
+  arch_isolated       = trimsuffix(var.system, "-linux")
+  arch                = local.arch_isolated == "aarch64" ? "arm64" : local.arch_isolated
+  nixos_ami_acc_id    = "080433136561" # Pretty sure all published AMIs so far have been from this account
+
+  # This is the biggest issue with this method as this just grabs the newest release-like AMI.
+  # Small tradeoff for a terraform native impl
```

Comment:
If you want we could remove support for `"latest"` since it's not a good practice to track the latest release anyway.  Instead, just make the release argument required